bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/AbstractComponentImpl.java

```diff
@@ -50,4 +73,44 @@ public String getId() {
     public String getExportedType() {
         return resource.getResourceType();
     }
+
+    /**
+     * Returns an auto generated component ID.
+     *
+     * The ID is an SHA hash of the component path, prefixed with the component name.
+     *
+     * If the component is referenced, the path is taken to be a concatenation of the component path,
+     * with the path of the first parent context resource that exists on the page or in the template.
+     * This ensures the ID is unique if the same component is referenced multiple times
+     * on the same page or template.
+     *
+     * @return the auto generated component ID
+     */
+    private String generateId() {
+        String resourceType = resource.getResourceType();
+        String prefix = StringUtils.substringAfterLast(resourceType, "/");
+        String path = resource.getPath();
+        PageManager pageManager = currentPage.getPageManager();
+        Page containingPage = pageManager.getContainingPage(resource);
+        Template template = currentPage.getTemplate();
+        Boolean inCurrentPage = (containingPage != null && StringUtils.equals(containingPage.getPath(), currentPage.getPath()));
+        Boolean inTemplate = (template != null && path.startsWith(template.getPath()));
```

Comment:
check for path != null

Reply:
Path here is from `Resource#getPath`, which isn't nullable so I think we're safe?